src/resources/MachineLearning/MachineLearningInterfaces.ts

```diff
@@ -96,9 +96,9 @@ export interface RegistrationModel extends GranularResource {
     versionMatcher?: string;
 }
 
-export interface MLModelCreated extends MLModel {
+export interface MLModelCreated extends Omit<MLModel, 'id'> {
```

Comment:
We could use the `New` type from BaseInterfaces here. Something like this (please double check writing this on top of my head 😛).

```ts
export interface MLModelCreated extends New<MLModel, 'modelId'> {
```

Reply:
It looks like the New type is usually used for parameter types, so I feel a bit relunctant to use it here, I think it's fine to use Omit.

Reply:
Interesting, are there other objects where this pattern applies?

Reply:
Ahh I see -- done.